nifi-commons/nifi-utils/src/main/java/org/apache/nifi/stream/io/util/TextLineDemarcator.java

```diff
@@ -159,8 +159,11 @@ private int isEol(byte currentByte, int currentIndex) {
                 this.index = currentIndex + 1;
                 this.fill();
             }
-            currentByte = this.buffer[currentIndex + 1];
-            crlfLength = currentByte == '\n' ? 2 : 1;
+            crlfLength = 1;
+            if (this.bufferLength != -1) {
```

Comment:
Hmm, why? Basically after the above call to _fill()_ it will be -1 or . . . Am I missing your point?